File: src/Module/Employee/EmployeeEntity.java
package Module.Employee;

import Module.File.FileEntity;
import Module.File.FileModel;
import Module.Group.GroupEntity;
import Module.Group.GroupModel;
import Module.Team.TeamEntity;
import Module.Team.TeamModel;
import Module.Ticket.TicketEntity;
import Module.Ticket.TicketModel;
import Module.TicketRead.TicketReadEntity;
import Module.TicketRead.TicketreadModel;
import Module.TicketRelater.TicketRelaterEntity;
import Module.TicketRelater.TicketrelaterModel;
import Module.TicketThread.TicketThreadEntity;
import Module.TicketThread.TicketthreadModel;
import Module.User.UserEntity;
import Module.User.UserModel;

import java.io.Serializable;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class EmployeeEntity implements Serializable {
    public int id;
    public String email;
    public String name;
    public int teamId;
    public int groupId;
    public String briefName;
    public TeamEntity teamEntity;
    public GroupEntity groupEntity;
    public List<FileEntity> fileEntities;
    public List<GroupEntity> groupEntities;
    public List<TeamEntity> teamEntities;
    public List<TicketEntity> createTicketEntities;
    public List<TicketEntity> assignTicketEntities;
    public List<TicketReadEntity> ticketReadEntities;
    public List<TicketRelaterEntity> ticketRelaterEntities;
    public List<TicketThreadEntity> ticketThreadEntities;
    public List<UserEntity> userEntities;

    public EmployeeEntity() {
    }


    public EmployeeEntity(EmployeeModel EmployeeModel, Object... objects) {
        this.id = EmployeeModel.getId();
        this.email = EmployeeModel.getEmail();
        this.name = EmployeeModel.getName();
        this.teamId = EmployeeModel.getTeamId();
        this.groupId = EmployeeModel.getGroupId();
        this.briefName = EmployeeModel.getBriefName();
        for (Object object : objects) {
            if (object instanceof TeamModel) {
                this.teamEntity = new TeamEntity((TeamModel) object);
            } else if (object instanceof GroupModel) {
                this.groupEntity = new GroupEntity((GroupModel) object);
            } else if (object instanceof Collection) {
                for (Object o : (Collection<Object>) object) {
                    if (o instanceof FileModel) {
                        this.fileEntities = ((Collection<FileModel>) object).parallelStream().map(FileEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof GroupModel) {
                        this.groupEntities = ((Collection<GroupModel>) object).parallelStream().map(GroupEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TeamModel) {
                        this.teamEntities = ((Collection<TeamModel>) object).parallelStream().map(TeamEntity::new).collect(Collectors.toList());
                        break;
                    } else if (o instanceof TicketModel) {
                        this.assignTicketEntities = ((Collection<TicketModel>) object).parallelStream().map(TicketEntity::new).collect(Collectors.toList());
                        break;
                    }else if (o instanceof TicketreadModel) {
                        this.ticketReadEntities = ((Collection<TicketreadModel>) object).parallelStream().map(TicketReadEntity::new).collect(Collectors.toList());
                        break;
                    }else if (o instanceof TicketrelaterModel) {
                        this.ticketRelaterEntities = ((Collection<TicketrelaterModel>) object).parallelStream().map(TicketRelaterEntity::new).collect(Collectors.toList());
                        break;
                    }else if (o instanceof TicketthreadModel) {
                        this.ticketThreadEntities = ((Collection<TicketthreadModel>) object).parallelStream().map(TicketThreadEntity::new).collect(Collectors.toList());
                        break;
                    }else if (o instanceof UserModel) {
                        this.userEntities = ((Collection<UserModel>) object).parallelStream().map(UserEntity::new).collect(Collectors.toList());
                        break;
                    }
                }
            }
        }

    }

    public EmployeeModel toModel() {
        EmployeeModel employeeModel = new EmployeeModel();
        employeeModel.setBriefName(briefName);
        employeeModel.setEmail(email);
        employeeModel.setName(name);
        employeeModel.setId(id);
        employeeModel.setGroupId(groupId);
        return employeeModel;
    }
}

File: web/Angular4/src/app/Modules/Employee/Employee.Entity.ts
import {TeamEntity} from "../Team/Team.Entity";
import {LocationEntity} from "../Location/Location.Entity";
import {FileEntity} from "../File/File.Entity";
import {TicketEntity} from "../Ticket/Ticket.Entity";
import {TicketReadEntity} from "../TicketRead/TicketRead.Entity";
import {TicketRelaterEntity} from "../TicketRelater/TicketRelater.Entity";
import {TicketThreadEntity} from "../TicketThread/TicketThread.Entity";
import {UserEntity} from "../User/User.Entity";

export class EmployeeEntity {

    id: number;
    email: string;
    name: string;
    teamId: number;
    locationId : number;
    briefName : string;
    teamEntity : TeamEntity;
    locationEntity : LocationEntity;
    fileEntities : FileEntity[];
    locationEntities : LocationEntity[];
    teamEntities : TeamEntity[];
    createTicketEntities : TicketEntity[];
    assignTicketEntities : TicketEntity[];
    ticketReadEntities : TicketReadEntity[];
    ticketRelaterEntities : TicketRelaterEntity[];
    ticketThreadEntities : TicketThreadEntity[];
    userEntities : UserEntity[];
    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;

    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.email = null;
            this.name = null;
            this.teamId = null;
            this.locationId = null;
            this.briefName = null;
            this.teamEntity = null;
            this.locationEntity = null;
            this.fileEntities = [];
            this.locationEntities = [];
            this.teamEntities = [];
            this.createTicketEntities = [];
            this.assignTicketEntities = [];
            this.ticketReadEntities = [];
            this.ticketRelaterEntities = [];
            this.ticketThreadEntities = [];
            this.userEntities = [];
        } else {
            this.id = data.id;
            this.email = data.email;
            this.name = data.name;
            this.teamId = data.teamId;
            this.locationId = data.locationId;
            this.briefName = data.briefName;
            this.teamEntity = data.teamEntity;
            this.locationEntity = data.locationEntity;
            if (data.fileEntities != null) {
                this.fileEntities = [];
                for (let item of data.fileEntities) {
                    this.fileEntities.push(new FileEntity(item));
                }
            } else data.fileEntities = [];
            if (data.locationEntities != null) {
                this.locationEntities = [];
                for (let item of data.locationEntities) {
                    this.locationEntities.push(new LocationEntity(item));
                }
            } else data.locationEntities = [];
            if (data.teamEntities != null) {
                this.teamEntities = [];
                for (let item of data.teamEntities) {
                    this.teamEntities.push(new TeamEntity(item));
                }
            } else data.teamEntities = [];
            if (data.createTicketEntities != null) {
                this.createTicketEntities = [];
                for (let item of data.createTicketEntities) {
                    this.createTicketEntities.push(new TicketEntity(item));
                }
            } else data.createTicketEntities = [];
            if (data.assignTicketEntities != null) {
                this.assignTicketEntities = [];
                for (let item of data.assignTicketEntities) {
                    this.assignTicketEntities.push(new TicketEntity(item));
                }
            } else data.assignTicketEntities = [];
            if (data.ticketReadEntities != null) {
                this.ticketReadEntities = [];
                for (let item of data.ticketReadEntities) {
                    this.ticketReadEntities.push(new TicketReadEntity(item));
                }
            } else data.ticketReadEntities = [];
            if (data.ticketRelaterEntities != null) {
                this.ticketRelaterEntities = [];
                for (let item of data.ticketRelaterEntities) {
                    this.ticketRelaterEntities.push(new TicketRelaterEntity(item));
                }
            } else data.ticketRelaterEntities = [];
            if (data.ticketThreadEntities != null) {
                this.ticketThreadEntities = [];
                for (let item of data.ticketThreadEntities) {
                    this.ticketThreadEntities.push(new TicketThreadEntity(item));
                }
            } else data.ticketThreadEntities = [];
            if (data.userEntities != null) {
                this.userEntities = [];
                for (let item of data.userEntities) {
                    this.userEntities.push(new UserEntity(item));
                }
            } else data.userEntities = [];
            if (this.teamEntity == null) this.teamEntity = new TeamEntity();
            if (this.locationEntity == null) this.locationEntity = new LocationEntity();
        }
        this.IsEdit = false;
    }
}